src/frontend/src/pages/DesignReviewView.tsx

```diff
@@ -0,0 +1,186 @@
+        {Array.from({ length: 6 }, (_, i) => (
+          <Box
+            sx={{
+              width: '1.5vw',
+              height: '1.5vw',
+              backgroundColor: colors[i]
+            }}
+          />
+        ))}
+        <Typography style={{ marginLeft: '10px', fontFamily: 'oswald' }}>
+          {totalUsers}/{totalUsers}
+        </Typography>
+      </Grid>
+    );
+  };
+
+  const renderAvailabilites = () => {
+    const fontSize = totalUsers > 10 ? '1em' : totalUsers > 15 ? '0.8em' : '1.2em';
+    return (
+      <Grid
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          backgroundColor: '#2C2C2C',
+          padding: '20px',
+          borderRadius: '8px',
+          height: '100%',
+          overflow: 'auto'
+        }}
+      >
+        <Grid style={{ display: 'flex', flexDirection: 'column', flexGrow: 1, marginBottom: '10px' }}>
+          {renderLegend()}
+          <Grid style={{ display: 'flex', gap: '2', marginTop: '10px' }}>
+            <Grid>
+              <Typography
+                style={{
+                  textDecoration: 'underline',
+                  fontFamily: 'oswald',
+                  fontSize: '1.5em',
+                  textAlign: 'center',
+                  marginBottom: '10px'
+                }}
+              >
+                Available Users
+              </Typography>
+              {currentAvailableUsers.map((user) => (
+                <Typography style={{ fontFamily: 'oswald', textAlign: 'center', fontSize }}>
+                  {user.firstName} {user.lastName}
+                </Typography>
+              ))}
+            </Grid>
+            <Grid>
+              <Typography
+                style={{
+                  textDecoration: 'underline',
+                  fontFamily: 'oswald',
+                  fontSize: '1.5em',
+                  textAlign: 'center',
+                  marginBottom: '10px'
+                }}
+              >
+                Unvailable Users
+              </Typography>
+              {currentUnavailableUsers.map((user) => (
+                <Typography style={{ fontFamily: 'oswald', textAlign: 'center', fontSize }}>
+                  {user.firstName} {user.lastName}
+                </Typography>
+              ))}
+            </Grid>
+          </Grid>
+          <Grid
+            style={{
+              marginTop: 'auto',
+              display: 'flex',
+              justifyContent: 'flex-end',
+              gap: '10px'
+            }}
+          >
+            <WarningIcon style={{ color: 'yellow', fontSize: '2em', marginTop: '5px' }} />
+            <NERFailButton>Cancel</NERFailButton>
+            <NERSuccessButton>Finalize</NERSuccessButton>
+          </Grid>
+        </Grid>
+      </Grid>
+    );
+  };
+
+  return (
+    <Grid display={'flex'}>
+      {renderSchedule()}
+      <Grid>{renderAvailabilites()}</Grid>
+    </Grid>
```

Comment:
I would make renderSchedule and renderAvailibilites their own components that are just called in this file

Reply:
also for this Grid component, you should either use the xs, md, container/ item functionality or just make the grid components Boxes.

Reply:
I can go through and leave more specific comments tmrw, but you're using Grid and Flexbox (`display: flex`) in a lot of places in this file. A lot of times you can hone in on which is actually giving you the effect you're getting and just use that one. Grid is best for positioning things in a row/ changing what's on what row for different screen sizes. Flexbox is best for positioning things in relation to each other inside a box (e.g. at regular intervals with a gap in between them or at either end of a box).

Reply:
For reference:
https://mui.com/material-ui/react-grid/
If you've never played around with this, it can be very helpful for seeing the different use cases of a flex box: https://flexboxfroggy.com/